File: internal/service/flow_version_service.go
package service

import (
	"github.com/nexus-flow/nexus/internal/model"
	"github.com/nexus-flow/nexus/internal/repository"
)

type FlowVersionDTO struct {
	ID                int64  `json:"id"`
	FlowName          string `json:"flowName,omitempty"`
	FlowVersion       string `json:"flowVersion"`
	TriggerURL        string `json:"triggerUrl,omitempty"`
	FlowVersionStatus int    `json:"flowVersionStatus"`
	FlowVersionRemark string `json:"flowVersionRemark,omitempty"`
}

func UpdateFlowVersionStatus(id int64, status int) error {
	return repository.UpdateFlowVersionStatus(id, status)
}

func DeleteFlowVersion(id int64) error {
	return repository.DeleteFlowVersion(id)
}

func GetLatestVersion(flowKey string) (string, error) {
	return repository.GetLatestVersion(flowKey)
}

func PageFlowVersions(p *model.PageParam, flowID *int64, status *int) ([]FlowVersionDTO, int64, error) {
	list, total, err := repository.PageFlowVersions(p, flowID, status)
	if err != nil {
		return nil, 0, err
	}
	dtos := make([]FlowVersionDTO, 0, len(list))
	for _, fv := range list {
		dtos = append(dtos, FlowVersionDTO{
			ID:                fv.ID,
			FlowVersion:       fv.FlowVersion,
			FlowVersionStatus: fv.FlowVersionStatus,
			FlowVersionRemark: fv.FlowVersionRemark,
		})
	}
	return dtos, total, nil
}

File: server/src/main/java/com/javaoffers/nexus/core/service/FlowVersionService.java
package com.javaoffers.nexus.core.service;

import com.javaoffers.nexus.core.model.FlowInfo;
import com.javaoffers.nexus.core.model.FlowVersion;
import com.javaoffers.nexus.core.model.PageParam;
import com.javaoffers.nexus.core.repository.FlowInfoMapper;
import com.javaoffers.nexus.core.repository.FlowVersionMapper;
import lombok.Data;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

/** 对应 Go core/service/flow_version_service.go。 */
@Service
public class FlowVersionService {

    private final FlowVersionMapper flowVersionMapper;
    private final FlowInfoMapper flowInfoMapper;

    public FlowVersionService(FlowVersionMapper flowVersionMapper, FlowInfoMapper flowInfoMapper) {
        this.flowVersionMapper = flowVersionMapper;
        this.flowInfoMapper = flowInfoMapper;
    }

    @Data
    public static class FlowVersionDTO {
        private Long id;
        private String flowVersion;
        private int flowVersionStatus;
        private String flowVersionRemark;
    }

    public void updateFlowVersionStatus(Long id, int status) {
        flowVersionMapper.updateStatus(id, status);
    }

    public void deleteFlowVersion(Long id) {
        flowVersionMapper.delete(id);
    }

    public String getLatestVersion(String flowKey) {
        FlowInfo info = flowInfoMapper.getByKey(flowKey);
        if (info == null) {
            return "v1";
        }
        long count = flowVersionMapper.countByFlowId(info.getId());
        return "v" + (count + 1);
    }

    public Page<FlowVersionDTO> pageFlowVersions(PageParam p, Long flowID, Integer status) {
        int pn = p.getPageNum() <= 0 ? 1 : p.getPageNum();
        int ps = p.getPageSize() <= 0 ? 10 : p.getPageSize();
        List<FlowVersion> list = flowVersionMapper.page(pn, ps, flowID, status);
        long total = flowVersionMapper.count(flowID, status);
        List<FlowVersionDTO> dtos = new ArrayList<>();
        for (FlowVersion fv : list) {
            FlowVersionDTO dto = new FlowVersionDTO();
            dto.setId(fv.getId());
            dto.setFlowVersion(fv.getFlowVersion());
            dto.setFlowVersionStatus(fv.getFlowVersionStatus());
            dto.setFlowVersionRemark(fv.getFlowVersionRemark());
            dtos.add(dto);
        }
        return new Page<>(total, dtos);
    }
}
